pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -0,0 +1,303 @@
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+//go:generate mockgen -source $GOFILE -package=$GOPACKAGE -destination=generated_mock_$GOFILE
+
+package network
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/vishvananda/netlink"
+
+	"kubevirt.io/kubevirt/pkg/log"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/network/dhcp"
+
+	lmf "github.com/subgraph/libmacouflage"
+)
+
+type VIF struct {
+	Name    string
+	IP      netlink.Addr
+	MAC     net.HardwareAddr
+	Gateway net.IP
+	Routes  *[]netlink.Route
+	Mtu     uint16
+}
+
+type NetworkHandler interface {
+	LinkByName(name string) (netlink.Link, error)
+	AddrList(link netlink.Link, family int) ([]netlink.Addr, error)
+	RouteList(link netlink.Link, family int) ([]netlink.Route, error)
+	AddrDel(link netlink.Link, addr *netlink.Addr) error
+	AddrAdd(link netlink.Link, addr *netlink.Addr) error
+	LinkSetDown(link netlink.Link) error
+	LinkSetUp(link netlink.Link) error
+	LinkAdd(link netlink.Link) error
+	ParseAddr(s string) (*netlink.Addr, error)
+	SetRandomMacAddrSameVendor(iface string) (net.HardwareAddr, error)
+	GetMacDetails(iface string) (net.HardwareAddr, error)
+	StartDHCP(nic *VIF, serverAddr *netlink.Addr)
+}
+
+type NetworkUtilsHandler struct{}
+
+var Handler NetworkHandler
+
+func (h *NetworkUtilsHandler) LinkByName(name string) (netlink.Link, error) {
+	return netlink.LinkByName(name)
+}
+func (h *NetworkUtilsHandler) AddrList(link netlink.Link, family int) ([]netlink.Addr, error) {
+	return netlink.AddrList(link, family)
+}
+func (h *NetworkUtilsHandler) RouteList(link netlink.Link, family int) ([]netlink.Route, error) {
+	return netlink.RouteList(link, family)
+}
+func (h *NetworkUtilsHandler) AddrDel(link netlink.Link, addr *netlink.Addr) error {
+	return netlink.AddrDel(link, addr)
+}
+func (h *NetworkUtilsHandler) LinkSetDown(link netlink.Link) error {
+	return netlink.LinkSetDown(link)
+}
+func (h *NetworkUtilsHandler) LinkSetUp(link netlink.Link) error {
+	return netlink.LinkSetUp(link)
+}
+func (h *NetworkUtilsHandler) LinkAdd(link netlink.Link) error {
+	return netlink.LinkAdd(link)
+}
+func (h *NetworkUtilsHandler) ParseAddr(s string) (*netlink.Addr, error) {
+	return netlink.ParseAddr(s)
+}
+func (h *NetworkUtilsHandler) AddrAdd(link netlink.Link, addr *netlink.Addr) error {
+	return netlink.AddrAdd(link, addr)
+}
+
+// GetMacDetails from an interface
+func (h *NetworkUtilsHandler) GetMacDetails(iface string) (net.HardwareAddr, error) {
+	currentMac, err := lmf.GetCurrentMac(iface)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get mac information for interface: %s", iface)
+		return nil, err
+	}
+	return currentMac, nil
+}
+
+// SetRandomMacAddrSameVendor changes the MAC address for a agiven interface
+func (h *NetworkUtilsHandler) SetRandomMacAddrSameVendor(iface string) (net.HardwareAddr, error) {
```

Comment:
Can we call that something like MoveMacToVirtualMachineInterface? Still not perfect and not describing everything but less cryptic. I would add the random mac to the description.

Reply:
Great!